src/sentry/testutils/helpers/features.py

```diff
@@ -103,8 +102,10 @@ def wrapped(self, *args, **kwargs):
 
 def apply_feature_flag_on_cls(feature_flag):
     def decorate(cls):
-        for name, fn in inspect.getmembers(cls, inspect.isfunction):
-            setattr(cls, name, with_feature(feature_flag)(fn))
+        # Wrap the [run](https://docs.python.org/3/library/unittest.html#unittest.TestCase.run) method
+        # with the with_feature decorator so all tests in the class can access the
+        # feature flag.
+        setattr(cls, "run", with_feature(feature_flag)(cls.run))
```

Comment:
This decorator was causing a lot of test flakes in this test_organization_tags [test](https://sentry.io/organizations/sentry/discover/results/?field=transaction&field=title&field=event.type&name=All+Events&project=2423079&query=%21pytest.result%3Askipped+transaction%3A%22pytest.runtest.protocol+tests%2Fsnuba%2Fapi%2Fendpoints%2Ftest_organization_tags.py%3A%3AOrganizationTagsTest%3A%3Atest_simple%22+title%3A%22AssertionError%3A+assert+%5B%7B%27key%27%3A+%27level%27%2C+%27name%27%3A+%27Level%27%2C+%27totalValues%27%3A+5%7D%2C+%7B%27key%27%3A+%27fruit%27%2C+%27name%27%3A+%27Fruit%27%2C+%27t...%22&sort=-event.type&statsPeriod=7d&widths=795&yAxis=count%28%29)

Wrapping every method in the class is probably causing something funky happen on teardown (haven't been able to track down the root cause) but wrapping just the run method with the decorator fixes the flake while still doing what we expect.